src/types/Slot0.sol

```diff
@@ -0,0 +1,112 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+struct Slot0 {
+    // the current price
+    uint160 sqrtPriceX96;
+    // the current tick
+    int24 tick;
+    // protocol fee, expressed in hundredths of a bip
+    // upper 12 bits are for 1->0, and the lower 12 are for 0->1
+    // the maximum is 1000 - meaning the maximum protocol fee is 0.1%
+    // the protocolFee is taken from the input first, then the lpFee is taken from the remaining input
+    uint24 protocolFee;
+    // used for the lp fee, either static at initialize or dynamic via hook
+    uint24 lpFee;
+}
+
+type Slot0Packed is bytes32;
+
+using Slot0Library for Slot0 global;
+using Slot0PackedLibrary for Slot0Packed global;
+
+/// @dev This library is used to read/write packed struct from storage to stack
+library Slot0Library {
+    function storePacked(Slot0 storage pointer, Slot0Packed _packed) internal {
+        /// @solidity memory-safe-assembly
+        assembly {
+            sstore(pointer.slot, _packed)
+        }
+    }
+
+    function loadPacked(Slot0 storage pointer) internal view returns (Slot0Packed _packed) {
+        /// @solidity memory-safe-assembly
+        assembly {
+            _packed := sload(pointer.slot)
+        }
+    }
+}
+
+/// @dev This library is used to get/set values in packed structure on stack
+library Slot0PackedLibrary {
+    uint256 private constant UINT160_MASK = 0x00FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF;
+    uint256 private constant INT24_MASK = 0xFFFFFF;
+    uint256 private constant UINT24_MASK = 0xFFFFFF;
+
+    uint256 private constant TICK_OFFSET = 160;
+    uint256 private constant PROTOCOL_FEE_OFFSET = 184;
+    uint256 private constant LP_FEE_OFFSET = 208;
```

Comment:
```suggestion
    uint8 private constant TICK_OFFSET = 160;
    uint8 private constant PROTOCOL_FEE_OFFSET = 184;
    uint8 private constant LP_FEE_OFFSET = 208;
```

Reply:
Great. That means solc is smart enough to use `PUSH1` for these instead of `PUSH32` so bytecode size is already optimized!